extras/ci/analytics/ci_visualization.py

```
@@ -41,19 +40,62 @@ def parse_args():
     return parser.parse_args()
 
 
+def validate_config(config):
+    """Validate expected config structure."""
+    if not isinstance(config, dict):
+        raise ValueError("Config root must be an object.")
```

Ruff (0.15.2)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)